pkg/ans/ans_test.go

```diff
@@ -0,0 +1,174 @@
+package ans
+
+import (
+	"encoding/json"
+	"github.com/SAP/jenkins-library/pkg/xsuaa"
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+type Examinee struct {
+	xsuaa     *xsuaa.XSUAA
+	server    *httptest.Server
+	ans       *ANS
+	onRequest func(rw http.ResponseWriter, req *http.Request)
+}
+
+func (e *Examinee) request(rw http.ResponseWriter, req *http.Request) {
+	e.onRequest(rw, req)
+}
+
+func (e *Examinee) finish() {
+	if e.server != nil {
+		e.server.Close()
+		e.server = nil
+	}
+}
+
+func (e *Examinee) init() {
+	if e.xsuaa == nil {
+		e.xsuaa = &xsuaa.XSUAA{
+			OAuthURL:     "https://my.test.oauth.provider",
+			ClientID:     "myTestClientID",
+			ClientSecret: "super secret",
+			CachedAuthToken: xsuaa.AuthToken{
+				TokenType:   "bearer",
+				AccessToken: "1234",
+				ExpiresIn:   12345,
+			},
+		}
+	}
+	if e.server == nil {
+		e.server = httptest.NewServer(http.HandlerFunc(e.request))
+	}
+	if e.ans == nil {
+		e.ans = &ANS{XSUAA: *e.xsuaa, URL: e.server.URL}
+	}
+}
+
+func (e *Examinee) execute(event Event, onRequest func(rw http.ResponseWriter, req *http.Request)) error {
+	e.init()
+	e.onRequest = onRequest
+	return e.ans.Send(event)
+}
+
+func TestANS_Send(t *testing.T) {
+	examinee := Examinee{}
+	defer examinee.finish()
+	examinee.init()
```

Comment:
@rodibrin I like it and adopted it. I would like it even more if we could have the ans.Send being executed in the test runs, to make it clear that this is what is tested. Currently it is hidden in the execute function. What do you think Roland?

Reply:
Yes, that is nice. I used that